### PR TITLE
use preprocessed t and xy instead of as_list_with_timestamps

### PR DESCRIPTION
**Pull request by tybug**
should be identical (though I haven't tested, we need to add a test case for aim correction). This should also be done for ur calculation so we can finally be rid of `as_list_with_timestamps`.

### CODE REVIEW

File: tests/test.py [hunk 1]
```diff
@@ -133,6 +134,30 @@ def test_num_invariance(self):
...
+        cls.r1 = ReplayPath(RES / "corrected_replay1.osr")
+
+    def test_cheated_replay(self):
+        r = list(self.cg.run(Check([self.r1], CorrectionDetect())))[0]
+        snaps = r.snaps
+
+        self.assertEqual(len(snaps), 15)
+        # beginning
+        self.assertEqual(snaps[0].time, 5103)
+        self.assertEqual(snaps[0].angle, 7.384911923724048)
```
Comment (InvisibleSymbol) [inline tests/test.py:150]: how about using `assertAlmostEqual`?

Comment (tybug) [reply tests/test.py:150]: yep

File: tests/test.py [hunk 2]
```diff
@@ -2,8 +2,9 @@
 from unittest import TestCase, skip, TestSuite, TextTestRunner
 from pathlib import Path
 import warnings
-from circleguard import (Circleguard, Check, ReplayMap, ReplayPath, RelaxDetect, StealDetect,
+from circleguard import (Circleguard, Check, ReplayMap, ReplayPath, RelaxDetect, StealDetect, CorrectionDetect,
                          RatelimitWeight, set_options, Map, User, MapUser, Mod, Loader, InvalidKeyException)
+from circleguard.investigator import Snap
```
Comment (InvisibleSymbol) [inline tests/test.py:7]: unused import

Comment (tybug) [reply tests/test.py:7]: oh right

**Pull request merged**